src/Microsoft.OData.Core/BindingPathHelper.cs

```diff
@@ -53,19 +54,19 @@ public static bool MatchBindingPath(IEdmPathExpression bindingPath, List<ODataPa
                 else if (segment is TypeSegment)
                 {
                     // May need match type if the binding path contains type cast.
-                    if (pathIndex >= 0 && paths[pathIndex].Contains("."))

```

Comment:
Contains(...) vs IndexOf(...) ? What's the different? 
I prefer to "Contains(...)", it's more readable, maintainable.

Reply:
IndexOf is more performant. I verified it using a console app and measured the numbers.